stage0/run.go

```diff
@@ -152,8 +153,8 @@ func generatePodManifest(cfg PrepareConfig, dir string) ([]byte, error) {
 		if pm.Apps.Get(*appName) != nil {
 			return fmt.Errorf("error: multiple apps with name %s", am.Name)
 		}
-		if am.App == nil {
-			return fmt.Errorf("error: image %s has no app section", img)
+		if am.App == nil && app.Exec == "" {
+			return fmt.Errorf("error: image %s has no app section and --exec argument not provided", img)
```

Comment:
Not strictly necessary here. Actually, I'd prefer without it.

On Fri, Sep 18, 2015 at 6:38 AM, Krzesimir Nowak <notifications@github.com>
wrote:

> In stage0/run.go
> <https://github.com/coreos/rkt/pull/1427#discussion_r39855005>:
>
> > @@ -152,8 +153,8 @@ func generatePodManifest(cfg PrepareConfig, dir string) ([]byte, error) {
> >  		if pm.Apps.Get(*appName) != nil {
> >  			return fmt.Errorf("error: multiple apps with name %s", am.Name)
> >  		}
> > -		if am.App == nil {
> > -			return fmt.Errorf("error: image %s has no app section", img)
> > +		if am.App == nil && app.Exec == "" {
> > +			return fmt.Errorf("error: image %s has no app section and --exec argument not provided", img)
>
> "and --exec argument *is* not provided"
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/coreos/rkt/pull/1427/files#r39855005>.
>